src/lib/optimizer/strategy/nested_expression_rule.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "all_type_variant.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class ProjectionNode;
+class LQPExpression;
+class LQPColumnReference;
+
+/**
+ * This optimizer rule looks for ProjectionNodes that contain calculable Expressions.
```

Comment:
"This optimizer rule looks for Expressions in ProjectionNodes that are calculable at planning time and replaces them with a constant"?